apps/coscrad-frontend/src/utils/generic-components/presenters/tables/generic-index-table-presenter/index-table.tsx

```diff
@@ -59,14 +59,14 @@ export const IndexTable = <T extends IBaseViewModel>({
     const [searchValue, setSearchValue] = useState('');
 
     // SEARCH LOGIC
-    const filteredTableData =
-        searchValue.length === 0
-            ? tableData
-            : tableData.filter((row) =>
-                  filterableProperties.some((propertyKey) =>
-                      String(row[propertyKey]).toLowerCase().includes(searchValue.toLowerCase())
-                  )
-              );
+
+    const [selectedFilterProperty, setSelectedFilterProperty] = useState(filterableProperties[0]);
```

Comment:
You  might consider making the state `selectedFilterProperties` instead. Then if they choose one of the 'single properties' (e.g. termEnglish) to filter by, store an array with just this property name. If they choose 'search all', make the array all of the property names. This is more extensible.